app/services/claim_service.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+require_dependency 'evss/claims_service'
+
+class ClaimService
+  EVSS_CLAIM_KEYS = %w(openClaims historicalClaims).freeze
+
+  def initialize(user)
+    @user = user
+  end
+
+  def all
+    raw_claims = client.all_claims.body
+    EVSS_CLAIM_KEYS.each_with_object([]) do |key, claims|
+      next unless raw_claims[key]
+      claims << raw_claims[key].map do |raw_claim|
+        find_or_initialize_claim(raw_claim)
+      end
+    end.flatten
+  rescue Faraday::Error::TimeoutError, Timeout::Error
+    Claim.where("data->>'participant_id' = ?", @user.participant_id)
+  end
+
+  def find_by_evss_id(id)
+    raw_claim = client.find_claim_by_id(id).body.fetch('claim', {})
+    find_or_initialize_claim(raw_claim)
+  rescue Faraday::Error::TimeoutError, Timeout::Error
```

Comment:
@austin-martinez so we're treating anything that involves user specified input as an error that we provide JSON feedback around, and anything related to downtime, or timeout issues as server 500 errors. We will log those as the actual error, but show a generic 500 error to the user. To the extent possible, you'll want to identify the "minor codes" or errors that can be returned from your underlying backend and come up with an appropriate translation to what we would show as our API. You can see examples of how we do this in Rx with ClientResponseError, though that still needs to be cleaned up and start using the errors I more recently created in VA-API-Common, i may actually pull those directly into vets-api. To my knowledge that gem is not being used by any other repo at present.